tchannel-core/src/main/java/com/uber/tchannel/api/SubChannel.java

```diff
@@ -225,4 +227,28 @@ protected ResponseRouter prepare(
     ) throws TChannelError {
         return send(request, null, 0);
     }
+
+    public <V> ListenableFuture<V> createError(Request request, ErrorType errorType, Throwable throwable) {
+        TFuture<V> future = TFuture.create();
+        ArgScheme argScheme = request.getArgScheme();
+        Response response = Response.build(argScheme, new ErrorResponse(
```

Comment:
maybe response builder should take care of building different Response based on different ArgScheme?